src/mme-app/actionHandlers/serviceRequestActionHandlers.cpp

```diff
@@ -270,64 +270,186 @@ ActStatus ActionHandlers::process_init_ctxt_resp_svc_req(ControlBlock& cb)
-        mb_msg.ue_idx = ue_ctxt->getContextID();
+    log_msg(LOG_DEBUG, "Leaving send_mb_req_to_sgw_svc_req \n");
 
-        memset(mb_msg.indication, 0, S11_MB_INDICATION_FLAG_SIZE); /*TODO : future*/
-        BearerContext* bearerCtxt = sessionCtxt->getBearerContext();
-        mb_msg.bearer_id = bearerCtxt->getBearerId();
+    return actStatus;
+}
 
-        memcpy(&(mb_msg.s11_sgw_c_fteid), &(sessionCtxt->getS11SgwCtrlFteid().fteid_m),
-                sizeof(struct fteid));
+/***************************************
+ * Action handler : process_mb_resp_svc_req
+ ***************************************/
+ActStatus ActionHandlers::process_mb_resp_svc_req(ControlBlock &cb)
+{
+    log_msg(LOG_DEBUG, "Inside process_mb_resp_svc_req \n");
 
-        memcpy(&(mb_msg.s1u_enb_fteid), &(bearerCtxt->getS1uEnbUserFteid().fteid_m),
-                sizeof(struct fteid));
-	mb_msg.servingNetworkIePresent = false;
-	mb_msg.userLocationInformationIePresent = false;
+    ProcedureStats::num_of_processed_mb_resp++;
 
-        cmn::ipc::IpcAddress destAddr;
-        destAddr.u32 = TipcServiceInstance::s11AppInstanceNum_c;
+    UEContext *ue_ctxt = static_cast<UEContext*>(cb.getPermDataBlock());
+    if (ue_ctxt == NULL)
+    {
+        log_msg(LOG_INFO, "process_mb_resp_svc_req: ue context is NULL \n");
 
-        MmeIpcInterface &mmeIpcIf = static_cast<MmeIpcInterface&>(compDb.getComponent(MmeIpcInterfaceCompId));
-	mmeIpcIf.dispatchIpcMsg((char *) &mb_msg, sizeof(mb_msg), destAddr);
+        MmeContextManagerUtils::deleteUEContext(cb.getCBIndex());
+        return ActStatus::HALT;
+    }
 
-        ProcedureStats::num_of_mb_req_to_sgw_sent ++;
-        log_msg(LOG_DEBUG, "Leaving send_mb_req_to_sgw_svc_req \n");
+    MsgBuffer *msgBuf = static_cast<MsgBuffer*>(cb.getMsgData());
 
-        return ActStatus::PROCEED;
-}
+    if (msgBuf == NULL)
+        return ActStatus::ABORT;
 
-/***************************************
-* Action handler : process_mb_resp_svc_req
-***************************************/
-ActStatus ActionHandlers::process_mb_resp_svc_req(ControlBlock& cb)
-{
-   	log_msg(LOG_DEBUG, "Inside process_mb_resp_svc_req \n");
+    const gtp_incoming_msg_data_t *gtp_msg_data =
+            static_cast<const gtp_incoming_msg_data_t*>(msgBuf->getDataPointer());
+    const struct MB_resp_Q_msg &mbr_info =
+            gtp_msg_data->msg_data.MB_resp_Q_msg_m;
 
-	ProcedureStats::num_of_processed_mb_resp ++;
+    ActStatus actStatus = ActStatus::PROCEED;
 
-	MmeContextManagerUtils::deallocateProcedureCtxt(cb, serviceRequest_c);
+    MmeProcedureCtxt *procCtxt =
+            dynamic_cast<MmeProcedureCtxt*>(cb.getTempDataBlock());
 
-	return ActStatus::PROCEED;
+    if (procCtxt != NULL)
+    {
+        if (mbr_info.cause != GTPV2C_CAUSE_REQUEST_ACCEPTED)
+        {
+            log_msg(LOG_INFO, "process_mb_resp_svc_req: MB Response Failure\n");
+
+            procCtxt->setMmeErrorCause(s11MBRespFailure_c);
```

Comment:
What would be subscriber state in MME if we return failure.

Reply:
We are going to detach the UE in case of Modify Bearer Response failure.